x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -6,43 +6,51 @@
 
 import { SavedObject, SavedObjectsClientContract } from 'src/core/server/';
 import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../constants';
-import { AssetReference, Installation, KibanaAssetType } from '../../../types';
+import { AssetReference, Installation, KibanaAssetType, CallESAsCurrentUser } from '../../../types';
 import { installIndexPatterns } from '../kibana/index_pattern/install';
 import * as Registry from '../registry';
 import { getObject } from './get_objects';
 import { getInstallation } from './index';
+import { installTemplates } from '../elasticsearch/template/install';
+import { installPipelines } from '../elasticsearch/ingest_pipeline/install';
 
 export async function installPackage(options: {
   savedObjectsClient: SavedObjectsClientContract;
   pkgkey: string;
+  callCluster: CallESAsCurrentUser;
 }): Promise<AssetReference[]> {
-  const { savedObjectsClient, pkgkey } = options;
-
-  const installIndexPatternsPromise = installIndexPatterns(savedObjectsClient, pkgkey);
+  const { savedObjectsClient, pkgkey, callCluster } = options;
+  const registryPackageInfo = await Registry.fetchInfo(pkgkey);
 
-  const installAssetsPromise = installAssets({
+  const installKibanaAssetsPromise = installKibanaAssets({
```

Comment:
Note: We should discuss the order assets are installed in. In general I would argue the Elasticsearch ones as they are needed at ingest time. No need to change anything now.